# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [4]:
import pandas as pd
import numpy as np
import random

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [38]:
randomlist1_4 = []
for i in range(0,100):
    n = random.randint(1,4)
    randomlist1_4.append(n)
len(randomlist1_4)
ds14=pd.Series(randomlist1_4)
ds14
# alternate
# ds1=pd.Series(np.random.randint(1,4,100))
# ds1

0     2
1     2
2     1
3     1
4     1
     ..
95    2
96    4
97    4
98    2
99    4
Length: 100, dtype: int64

In [22]:
randomlist1_3 = []
for i in range(0,100):
    n = random.randint(1,3)
    randomlist1_3.append(n)
len(randomlist1_3)
ds13=pd.Series(randomlist1_3)
ds13

0     2
1     3
2     2
3     1
4     3
     ..
95    3
96    2
97    1
98    3
99    2
Length: 100, dtype: int64

In [23]:
randomlist1__3 = []
for i in range(0,100):
    n = random.randint(10000,30000)
    randomlist1__3.append(n)
len(randomlist1__3)
ds1__3=pd.Series(randomlist1__3)
ds1__3

0     12108
1     12701
2     11092
3     16365
4     20387
      ...  
95    18049
96    12783
97    24196
98    26366
99    10050
Length: 100, dtype: int64

### Step 3. Let's create a DataFrame by joinning the Series by column

In [26]:
dataframe=pd.DataFrame({
    "rand1-4":ds13,
    "rand1-3":ds14,
    "rand10000-30000":ds1__3
})
dataframe

rand1-4  rand1-3  rand10000-30000
0         2       27            12108
1         3       16            12701
2         2       30            11092
3         1       10            16365
4         3        3            20387
..      ...      ...              ...
95        3       21            18049
96        2        7            12783
97        1       28            24196
98        3       16            26366
99        2       29            10050

[100 rows x 3 columns]

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [42]:
dataframe.rename(columns = {'rand1-4':'bedrs','rand1-3':'bathrs','rand10000-30000':'price_sqr_meter'}, inplace = True)
# alternative approach
#dataframe.columns=['bedrs','bathrs','price_sqr_meter']
dataframe

bedrs  bathrs  price_sqr_meter
0       2      27            12108
1       3      16            12701
2       2      30            11092
3       1      10            16365
4       3       3            20387
..    ...     ...              ...
95      3      21            18049
96      2       7            12783
97      1      28            24196
98      3      16            26366
99      2      29            10050

[100 rows x 3 columns]

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [47]:

bigcolumn=pd.concat([ds14,ds13,ds1__3]).reset_index(drop=True)
bigcolumn.name="bigcolumn"
bigcolumn

0          2
1          2
2          1
3          1
4          1
       ...  
295    18049
296    12783
297    24196
298    26366
299    10050
Name: bigcolumn, Length: 300, dtype: int64

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [55]:
df99=pd.DataFrame(pd.concat([ds14,ds13,ds1__3]))
df99

0
0       2
1       2
2       1
3       1
4       1
..    ...
95  18049
96  12783
97  24196
98  26366
99  10050

[300 rows x 1 columns]

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [57]:
df99.reset_index(drop=True)

0
0        2
1        2
2        1
3        1
4        1
..     ...
295  18049
296  12783
297  24196
298  26366
299  10050

[300 rows x 1 columns]